atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStoreTest.java

```diff
@@ -0,0 +1,141 @@
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.timestamp.AbstractDbTimestampBoundStoreTest;
+import com.palantir.timestamp.MultipleRunningTimestampServiceError;
+import com.palantir.timestamp.TimestampBoundStore;
+
+public class CassandraTimestampBoundStoreTest extends AbstractDbTimestampBoundStoreTest {
+    private static final long CASSANDRA_TIMESTAMP = 0L;
+    private static final String ROW_AND_COLUMN_NAME = "ts";
+    public static final long NEW_LIMIT = 100;
+
+    @ClassRule
+    public static final Containers CONTAINERS = new Containers(CassandraTimestampIntegrationTest.class)
+            .with(new CassandraContainer());
+
+    private CassandraKeyValueService kv = CassandraKeyValueService.create(
+            CassandraKeyValueServiceConfigManager.createSimpleManager(CassandraContainer.KVS_CONFIG),
+            CassandraContainer.LEADER_CONFIG);
+
+    @Override
+    public TimestampBoundStore createTimestampBoundStore() {
+        return CassandraTimestampBoundStore.create(kv);
+    }
+
+    @Test
+    public void storeWithEmptyTableThrows() {
+        assertThatThrownBy(() -> store.storeUpperLimit(NEW_LIMIT + 1))
+                .isExactlyInstanceOf(Error.class);
+    }
+
+    @Test
+    public void canGetNewFormat() {
+        insertTimestampWithCorrectId(NEW_LIMIT + 1);
+        assertCorrect();
+    }
+
+    @Test
+    public void canGetOldFormat() {
+        insertTimestampOld(NEW_LIMIT + 1);
+        assertCorrect();
+    }
+
+    @Test
+    public void storeWithWrongTimestampCorrectIdSucceeds() {
+        long limit = store.getUpperLimit();
+        insertTimestampWithCorrectId(NEW_LIMIT);
+        assertThat(limit).isNotEqualTo(NEW_LIMIT);
+        store.storeUpperLimit(NEW_LIMIT + 1);
+        assertCorrect();
+    }
+
+    @Test
+    public void storeWithWrongTimestampNoIdSucceeds() {
+        long limit = store.getUpperLimit();
+        insertTimestampOld(NEW_LIMIT);
+        assertThat(limit).isNotEqualTo(NEW_LIMIT);
+        store.storeUpperLimit(NEW_LIMIT + 1);
+        assertCorrect();
+    }
+
+    @Test
+    public void storeWithWrongTimestampWrongIdThrows() {
+        long limit = store.getUpperLimit();
+        insertTimestampWithFakeId(NEW_LIMIT);
+        assertThat(limit).isNotEqualTo(NEW_LIMIT);
+        assertThatThrownBy(() -> store.storeUpperLimit(NEW_LIMIT + 1))
+                .isExactlyInstanceOf(MultipleRunningTimestampServiceError.class);
+    }
+
+    @Test
+    public void storeWithRightTimestampWrongIdThrows() {
+        long limit = store.getUpperLimit();
+        insertTimestampWithFakeId(limit);
+        assertThatThrownBy(() -> store.storeUpperLimit(NEW_LIMIT + 1))
+                .isExactlyInstanceOf(MultipleRunningTimestampServiceError.class);
+    }
+
+    @After
+    public void cleanUp() {
+        kv.dropTable(AtlasDbConstants.TIMESTAMP_TABLE);
+    }
+
+    private void assertCorrect() {
```

Comment:
nit: Not very clear what `Correct` means on its own, could we name this `assertUpperLimitEqualsNewLimitPlusOne` (or better, define another constant for NL+1 and use that)?